openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MixedLoadTest.java

```diff
@@ -63,13 +80,25 @@ public void execute(StfCoreExtension test) throws StfException {
 				.addProjectToClasspath("openjdk.test.nio")
 				.addProjectToClasspath("openjdk.test.concurrent")
 				.addJarToClasspath(mauveJar)
-				.addJarToClasspath(fileSystemJar)
-				.setAbortIfOutOfMemory(false)
+				.addJarToClasspath(fileSystemJar);
+				
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation.setTimeLimit(timelimitInMins);	// If it's a time based test, stop execution after given time duration
+		}
+		loadTestInvocation = loadTestInvocation.setAbortIfOutOfMemory(false)
 				.addSuite("mix")
-				.setSuiteThreadCount(cpuCount - 2, 10)	
-				.setSuiteNumTests(totalTests * 200) // Leave 1 cpu for the JIT. i for GC and set min 20			
-				.setSuiteInventory(inventoryFile) 			// Point at the file which lists the tests
-				.setSuiteRandomSelection();		  			// Randomly pick the next test each time
+				.setSuiteThreadCount(cpuCount - 2, 10); 
+		if (isTimeBasedLoadTest) { 
+			loadTestInvocation = loadTestInvocation
+				.setSuiteNumTests(totalTests * 80000000 * testCountMultiplier); 	// If it's a time based test, run a very large load
+		} else {  
+			loadTestInvocation = loadTestInvocation
+				.setSuiteNumTests(totalTests * 200);	// Leave 1 cpu for the JIT. i for GC and set min 20
+		}
+		
+		loadTestInvocation = loadTestInvocation
+				.setSuiteInventory(inventoryFile) 	// Point at the file which lists the tests
+				.setSuiteRandomSelection(); 		// Randomly pick the next test each time
 				
 		test.doRunForegroundProcess("Run mixed unit tests", "LT", Echo.ECHO_ON,
 				ExpectedOutcome.cleanRun().within(timeout), 
```

Comment:
`timeout` is hard coded to `3h`.  It should be adjusted for a time limited test.  The time limit is when stf stops submitting new tests, so they will need time to complete.  To achieve roughly the same as currently behavior `timeout` should be `timeout + timelimit`.

Reply:
Is this test ever going to be used to run a load that's 3 hour long (or longer)? The original requirement was to write a STF load test that runs for 5 minutes. That's why the original `3h` timeout was left as is. 

If we plan to run really large loads such as 3h, we need to first make sure that the workload set would last that long.  (the current workload here was only tested to last 20 minutes. It may last much longer-- but I haven't tested it past the 20 minute mark). 

That said, I've updated the test to use a timeout value of `default timeout (3h) + timeLimit`